resources/js/Layouts/MapLayout.tsx

```diff
@@ -1,9 +1,59 @@
-import { PropsWithChildren } from 'react';
+import { PropsWithChildren, useEffect, useRef } from 'react';
 import BottomNav from '@/Components/Shared/BottomNav';
 
 interface MapLayoutProps extends PropsWithChildren {}
 
+/**
+ * MapLayout component that prevents body scroll when active.
+ *
+ * Uses a ref-based instance counter instead of module-level globals
+ * to avoid HMR issues and ensure proper cleanup per application instance.
+ */
 export default function MapLayout({ children }: MapLayoutProps) {
+  // Track if this instance has applied overflow styles
+  const hasAppliedStyles = useRef(false);
+
+  // Store original styles to restore on unmount
+  const originalStyles = useRef<{
+    htmlOverflow: string;
+    bodyOverflow: string;
+    htmlOverscroll: string;
+    bodyOverscroll: string;
+  } | null>(null);
+
+  useEffect(() => {
+    const html = document.documentElement;
+    const body = document.body;
+
+    // Store original styles before modifying
+    originalStyles.current = {
+      htmlOverflow: html.style.overflow,
+      bodyOverflow: body.style.overflow,
+      htmlOverscroll: html.style.overscrollBehavior,
+      bodyOverscroll: body.style.overscrollBehavior,
+    };
+
+    // Apply overflow hidden to prevent body scroll on map page
+    html.style.overflow = 'hidden';
+    body.style.overflow = 'hidden';
+    html.style.overscrollBehavior = 'none';
+    body.style.overscrollBehavior = 'none';
+
+    hasAppliedStyles.current = true;
+
+    return () => {
+      // Restore original styles on unmount
+      if (hasAppliedStyles.current && originalStyles.current) {
+        html.style.overflow = originalStyles.current.htmlOverflow;
+        body.style.overflow = originalStyles.current.bodyOverflow;
+        html.style.overscrollBehavior = originalStyles.current.htmlOverscroll;
+        body.style.overscrollBehavior = originalStyles.current.bodyOverscroll;
+
+        hasAppliedStyles.current = false;
+      }
+    };
+  }, []);
+
   return (
```

Comment:
I would move this to styles, it would be cleaner, and have less potential for errors, for example:

```suggestion
/**
 * MapLayout component that prevents body scroll when active.
 *
 * Toggles 'is-map-active' class on html/body to handle overflow via CSS.
 */
export default function MapLayout({ children }: MapLayoutProps) {
  useEffect(() => {
    const html = document.documentElement;
    const body = document.body;

    // Apply class to prevent scroll
    html.classList.add('is-map-active');
    body.classList.add('is-map-active');

    return () => {
      // Remove class on unmount
      html.classList.remove('is-map-active');
      body.classList.remove('is-map-active');
    };
  }, []);

```

and add this to `_map.scss`:

```scss
body.is-map-active,
html.is-map-active {
  overflow: hidden !important;
  overscroll-behavior: none !important;
}
```